chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -781,21 +777,14 @@ impl NetworkState {
                 None
             }
             RoutedMessageBody::StatePartRequest(request) => {
-                let mut queue = self.tier3_requests.lock();
-                if queue.len() < LIMIT_TIER3_REQUESTS {
-                    queue.push_back(Tier3Request {
-                        peer_info: PeerInfo {
-                            id: peer_id,
-                            addr: Some(request.addr),
-                            account_id: None,
-                        },
-                        body: Tier3RequestBody::StatePart(StatePartRequestBody {
-                            shard_id: request.shard_id,
-                            sync_hash: request.sync_hash,
-                            part_id: request.part_id,
-                        }),
-                    });
-                }
+                self.peer_manager_adapter.send(Tier3Request {
```

Comment:
I'm curious why can't we directly send the StatePartRequest to the view client from here like we are handling all the other messages? What makes receiving a Tier3Request special here?

Reply:
The main difference is that we need to [create new connections](https://github.com/near/nearcore/pull/12353/files#diff-b40c686ecfc158bc56c26a76881529c88ad148956cb668453fa9a00da8660954R1306-R1312) to handle these requests. We cannot spawn other PeerActors from a PeerActor; it needs to be done by the PeerManagerActor.